src/rpc/mining.cpp

```diff
@@ -146,6 +147,48 @@ UniValue generate(const JSONRPCRequest& request)
     return blockHashes;
 }
 
+UniValue generatetoaddress(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 2 || request.params.size() > 3)
```

Comment:
it must throw if a third argument is given.
i.e.
```suggestion
    if (request.fHelp || request.params.size() != 2)
```

Reply:
fixed.